### PR TITLE
Implement twitter with resize event

### PR DESCRIPTION
**Pull request by indianburger**
Intent to implement reference: #3377
Twitter exposes a resize event which can be used instead of setting up listeners. Also use the element returned by `twttr.widgets.createTweet` instead of looking for iframes/shadow DOM elements. This avoid AMP breaking when twitter changes internal implementation.

Testing done: 
`gulp dist --fortesting; gulp test`: I have 2 unrelated tests that fail constantly. Perhaps I'm running it wrongly?
Manual tests. Ran the twitter example page in IE edge 12, mobile safari 9.2, Chrome 50, FF 46, Android Chrome 49. 

I can do more testing, let me know where I should spend my time.

### CODE REVIEW

**Comment by mkhatib** [comment]
This LGTM thanks for doing some manual testing. @cramforce mind taking a look as well?

File: 3p/twitter.js [hunk 1]
```diff
@@ -59,43 +59,19 @@ export function twitter(global, data) {
...
-          subtree: true,
-          childList: true,
-        });
-        measure();
-      } else {
-        console./*OK*/error('No iframe or shadow root found for tweet.');
+
+    let twitterWidgetSandbox;
+    twttr.events.bind('resize', event => {
+      if (twitterWidgetSandbox === event.target) {
```
Comment (cramforce) [inline 3p/twitter.js]: Maybe add a comment as to what other cases happen here.

**Comment by cramforce** [comment]
Nice! So much simpler :)

LGTM with one comment.

File: 3p/twitter.js [hunk 2]
```diff
@@ -59,43 +59,20 @@ export function twitter(global, data) {
...
+
+    let twitterWidgetSandbox;
+    twttr.events.bind('resize', event => {
+      // To be safe, make sure the resize event was triggered for the widget we created below.
+      if (twitterWidgetSandbox === event.target) {
+        resize(event.target);
+      }
+    });
+
+    twttr.widgets.createTweet(data.tweetid, tweet, data).then(el => {
```
Comment (mkhatib) [inline 3p/twitter.js]: Seems you still need to `/*OK*/then` this promise. You can check for failures locally with `gulp presubmit`

File: 3p/twitter.js [hunk 3]
```diff
@@ -59,43 +59,20 @@ export function twitter(global, data) {
...
-        });
-        measure();
-      } else {
-        console./*OK*/error('No iframe or shadow root found for tweet.');
+
+    let twitterWidgetSandbox;
+    twttr.events.bind('resize', event => {
+      // To be safe, make sure the resize event was triggered for the widget we created below.
+      if (twitterWidgetSandbox === event.target) {
+        resize(event.target);
```
Comment (jridgewell) [inline 3p/twitter.js]: Nit pick: mind using `twitterWidgetSandbox` since we already have the reference?

File: 3p/twitter.js (hunk 2 above)
Comment (cramforce) [inline 3p/twitter.js]: Yep. Quick explanation: We do not ship with a promise polyfill inside of this iframe ourselves.

**Comment by mkhatib** [comment]
Thanks @indianburger for sending this! Let us know if there are some more stuff we can do to make `amp-twitter` even better (e.g. other useful params to expose for publishers to use).

**Pull request merged**